pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+			r.status.RemoveDeployments(types.NamespacedName{Name: request.Name, Namespace: request.Namespace})
+			// Get the unready EGW. Let's say we have 2 EGW resources red and blue.
+			// Red has already degraded. When the user deletes Red, TigeraStatus should go back to available
+			// as Blue is healthy. If all the EGWs are ready, clear the degraded TigeraStatus.
+			// If at least one of the EGWs is unhealthy, get the degraded msg from the conditions and
+			// update the TigeraStatus.
+			unreadyEGW := getUnreadyEgressGateway(egws)
+			if unreadyEGW != nil {
+				r.status.SetDegraded(operatorv1.ResourceNotReady,
+					fmt.Sprintf("Error reconciling Egress Gateway resource. Name=%s Namespace=%s", unreadyEGW.Name, unreadyEGW.Namespace),
+					fmt.Errorf("%s", getDegradedMsg(unreadyEGW)), reqLogger)
+				return reconcile.Result{}, nil
+			}
+			r.status.ClearDegraded()
+			return reconcile.Result{}, nil
+		}
+		// If the EGW resource is present, reconcile only that resource.
+		// Remove this from the list of EGWs before computing status.
+		egwsToReconcile = []operatorv1.EgressGateway{*requestedEGW}
+		egws = append(egws[:idx], egws[idx+1:]...)
+	}
+	r.status.OnCRFound()
+
+	// Get the unready EGW.
+	unreadyEGW := getUnreadyEgressGateway(egws)
+
+	if !r.licenseAPIReady.IsReady() {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for LicenseKeyAPI to be ready", nil, reqLogger)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	installation, installStatus, err := utils.GetInstallationAndStatus(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Error(err, "Installation not found")
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
+			// Set the EGW resource's condition to Degraded.
+			for _, egw := range egwsToReconcile {
+				r.setDegraded(ctx, &egw, "Installation not found", err)
+			}
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Error querying installation")
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error querying installation", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	variant := installStatus.Variant
+	if variant != operatorv1.TigeraSecureEnterprise {
+		reqLogger.Error(err, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise))
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if installStatus.CalicoVersion != components.EnterpriseRelease {
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
```

Comment:
Seems like we should provide a reason why we're not doing anything. I'm not sure if Degraded is appropriate and I realize I don't think we have a way to set explicit Progressing and a reason.

Do we need to requeue here because status updates don't trigger a reconcile?

Reply:
It should be clear if the operator is stuck waiting for this condition, and shouldn't be silent as far as TigeraStatus.

Reply:
Agreed - degraded definitely isn't appropriate, but we should at a minimum log out a reason here explaining that we're waiting for the roll-out to occur, and probably should set a TigeraStatus message to explain that we're waiting for calico-node to finish upgrading.

Reply:
For now, I have just logged a message.

Reply:
@sridhartigera nit, but calico -> "Calico" since it's a proper noun.

Also, it would be good to include the version of Calico that we are waiting for in the log message.